src/App.js

```diff
@@ -43,10 +79,18 @@ class App extends React.Component {
 					<HeaderComp />
 					<Row>
 						<Col md={6}>
+							TEST
 							{free_week_rotation}
 						</Col>
 						<Col md={6}>
-							<SearchFieldComp username={this.state.username} changed_handler={this.changedHandler.bind(this)} />
+							<SearchFieldComp 
+								username={this.state.username} 
+								changed_handler={this.changedHandler.bind(this)}
+								submit_handler={this.searchForUser.bind(this)} />
+
```

Comment:
A component of this kind should handle it's own state and logic. Definitely not the app module.